conans/client/tools/apple.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import subprocess
+
+
+def cmd_output(command):
+    return subprocess.check_output(command).decode().strip()
+
+
+def is_apple_os(_os):
+    """returns True if OS is Apple one (Macos, iOS, watchOS or tvOS"""
+    return str(_os) in ['Macos', 'iOS', 'watchOS', 'tvOS']
+
+
+def to_apple_arch(arch):
+    """converts conan-style architecture into Apple-style arch"""
+    return {'x86': 'i386',
+            'x86_64': 'x86_64',
+            'armv7': 'armv7',
+            'armv8': 'arm64',
+            'armv7s': 'armv7s',
+            'armv7k': 'armv7k'}.get(str(arch))
+
+
+def apple_sdk_name(settings):
+    """returns proper SDK name suitable for OS and architecture
+    we're building for (considering simulators)"""
+    arch = settings.get_safe('arch')
+    _os = settings.get_safe('os')
+    if str(arch).startswith('x86'):
+        return {'Macos': 'macosx',
+                'iOS': 'iphonesimulator',
+                'watchOS': 'watchsimulator',
+                'tvOS': 'appletvsimulator'}.get(str(_os))
+    elif str(arch).startswith('arm'):
+        return {'iOS': 'iphoneos',
+                'watchOS': 'watchos',
+                'tvOS': 'appletvos'}.get(str(_os))
+    else:
+        return None
+
+
+def apple_deployment_target_env_name(_os):
```

Comment:
What are these variables used for? CMake related? XCode related?...

Reply:
in certain projects (e.g. log4cplus, ffmpeg) ./configure use them